src/Board.hs

```diff
@@ -24,7 +26,7 @@ data SlideError       = InvalidNumber | NumberNotAdjacentToBlank deriving (Show,
 instance Show Board where
     show b@(Board (Dimensions width height) _) =
         unlines $ map showRow [0..height-1]
-        where showRow y               = foldl (++) "" $ map (\x -> showPosition b $ Position x y) [0..width-1]
+        where showRow y               = concat $ map (\x -> showPosition b $ Position x y) [0..width-1]
```

Comment:
There's also a function in Data.List called `concatMap`, which does the same thing as `concat . map`. I'm not sure if people actually like to use it though; having the `concat` and `map` separate is probably clearer if most people don't know about the existence of `concatMap`.